### ISSUE TITLE
Serv3 needs type serializers for collections that uses other type serializers for validating its contents

### ISSUE THREAD

**Issue by gradientvera** [space-wizards/RobustToolbox#2084]
Instead of having `PrototypeIdListSerializer`, `PrototypeIdHashSetSerializer` and so on, you'd have generic serializers which take other serializers for that collection's contents.

Example: `ListSerializer<string, PrototypeIdSerializer<EntityPrototype>>` would serialize/validate `List<string>` the same way `PrototypeIdListSerializer<EntityPrototype>` would right now.

**Comment by gradientvera** [comment]
@PaulRitter @DrSmugleaf What do you think about this?

**Comment by PaulRitter** [comment]
yes this is definitely something we want to do. this was very much planned but then life hit
